source/httpfs/driver.go

```diff
@@ -10,10 +10,21 @@ import (
 	"github.com/golang-migrate/migrate/v4/source"
 )
 
-// driver is a migration source driver for reading migrations from
+// ErrDuplicateMigration is an error type for reporting duplicate migration
+// files.
+type ErrDuplicateMigration struct {
+	Filename string
+}
+
+// Error implements error interface.
+func (e ErrDuplicateMigration) Error() string {
+	return fmt.Sprintf("duplicate migration file: %s", e.Filename)
+}
+
+// Driver is a migration source driver for reading migrations from
 // http.FileSystem instances. It implements source.Driver interface and can be
```

Comment:
Update this comment since this shouldn't implement the `source.Driver` interface